tests/integration/share_access_test.go

```diff
@@ -44,34 +51,46 @@ func (s *ShareAccessSuite) SetupSuite() {
 // TestLogin verifies that users can log into the share.
 func (s *ShareAccessSuite) TestLogin() {
 	ctx := s.defaultContext()
+	require := s.Require()
 	tc := kube.NewTestClient("")
-	smbclient := smbclient.MustPodExec(tc, testNamespace, s.clientPod, "")
-	err := smbclient.CacheFlush(ctx)
-	s.Require().NoError(err)
+	client := smbclient.MustPodExec(tc, testNamespace, s.clientPod, "")
+	err := client.CacheFlush(ctx)
+	require.NoError(err)
+
+	ctx2, cancel := context.WithTimeout(ctx, loginTestTimeout)
+	defer cancel()
 	for _, auth := range s.auths {
-		err := smbclient.Command(
-			ctx,
-			s.share,
-			auth,
-			[]string{"ls"})
-		s.Require().NoError(err)
+		var cmderr error
+		err := poll.TryUntil(ctx2, &poll.Prober{
+			RetryInterval: loginTestInterval,
+			Cond: func() (bool, error) {
+				cmderr = client.Command(
+					ctx,
+					s.share,
+					auth,
+					[]string{"ls"})
+				return cmderr == nil, nil
+			},
+		})
+		require.NoError(cmderr)
```

Comment:
Do we really need to check `cmderr` for any errors? Isn't it implicitly covered under `err`?

Reply:
Added comment.